dbt/adapters/bigquery/relation.py

```diff
@@ -63,11 +68,47 @@ def get_include_policy(cls, relation, information_schema_view):
         if information_schema_view == "__TABLES__":
             identifier = False
 
+        # In the future, let's refactor so that location/region can also be a
+        # ComponentName, so that we can have logic like:
+        #
+        # region = False
+        # if information_schema_view == "OBJECT_PRIVILEGES":
+        #     region = True
+
         return relation.include_policy.replace(
             schema=schema,
             identifier=identifier,
         )
 
+    def get_region_identifier(self) -> str:
+        region_id = f"region-{self.location}"
+        return self.quoted(region_id)
+
+    @classmethod
+    def from_relation(cls, relation, information_schema_view):
+        info_schema = super().from_relation(relation, information_schema_view)
+        if information_schema_view == "OBJECT_PRIVILEGES":
+            # OBJECT_PRIVILEGES require a location.  If the location is blank there is nothing
+            # the user can do about it.
+            if not relation.location:
+                msg = f'No location found when trying to retrieve "{information_schema_view}"'
```

Comment:
Slight alternative you could accept or reject:
```suggestion
                msg = f'No location/region found when trying to retrieve "{information_schema_view}"'
```